src/components/template/TemplatePreviewPage.tsx

```diff
@@ -0,0 +1,293 @@
+import { Separator } from '@/components/ui/separator';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs';
+import { fetchTemplateById } from '@/libs/actions/templates';
+import { getClientSecret } from '@/libs/actions/user';
+import { useTemplateStore } from '@/libs/store/TemplateStore';
+import { generateCodeSnippets } from '@/service/generateCodeSnippets';
+import type { Template } from '@/types/Template';
+
+import TemplatePreviewLoading from './TemplatePreviewLoading';
+
+export default function TemplatePreviewPage() {
+  const { user } = useUser();
+  const [activeTab, setActiveTab] = useState('javascript');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [previewTemplate, setPreviewTemplate] = useState<Template | null>(null);
+  const [secret, setSecret] = useState<string>('');
+  const [copied, setCopied] = useState<string | null>(null);
+  const { selectedTemplate } = useTemplateStore();
+
+  const handleSecret = async () => {
+    if (!user) {
+      return;
+    }
+    try {
+      const result = await getClientSecret(user.id);
+
+      if (!result) {
+        return;
+      }
+
+      setSecret(result.clientSecret);
+    } catch (error) {
+      console.error(`Failed to retrieve client Secret, ${error}`);
+    }
+  };
+
+  useEffect(() => {
+    handleSecret();
+  }, [user]);
+
+  useEffect(() => {
+    // Simulate API call to fetch template data
+    const fetchTemplate = async () => {
+      setLoading(true);
+      try {
+        // In a real app, this would be an API call
+        const data = await fetchTemplateById(selectedTemplate as string);
+        setPreviewTemplate(data.data as Template);
+      } catch (error) {
+        console.error('Error fetching template:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchTemplate();
+  }, [selectedTemplate]);
+
+  const handleCopy = (code: string, language: string) => {
+    navigator.clipboard.writeText(code);
+    setCopied(language);
+    setTimeout(() => setCopied(null), 2000);
+  };
+
+  // Generate a formatted version of the sample data for code snippets
+  const formattedSampleData = previewTemplate?.templateSampleData
+    ? JSON.stringify(previewTemplate.templateSampleData, null, 2)
+    : `{}`;
+
+  // Code snippets for different languages
+  const codeSnippets = useMemo(() => {
+    return generateCodeSnippets({
+      BASE_API_URL,
+      templateId: previewTemplate?.templateId as string,
+      secret,
+      userId: user?.id as string,
+      formattedSampleData,
+    });
+  }, [BASE_API_URL, previewTemplate, secret, user, formattedSampleData]);
+
+  if (loading) {
+    return (
+      <TemplatePreviewLoading />
+    );
+  }
+
+  return (
+    <div className="container max-w-7xl py-6">
+      {/* Header */}
+      <div className="mb-6 flex flex-col justify-between gap-4 md:flex-row md:items-center">
```

Comment:
extract all the components in separate files

MainContent
PreviewContent 
etc

Reply:
maintain a folder structure like
components/template/preview
           - index.tsx
           - MainContent.tsx
           - TemplateInformation.tsx